app/controllers/poll_responses_controller.rb

```diff
@@ -25,10 +26,12 @@ def create
     else
       flash[:notice] = "Saving response failed"
     end
-    if request.xhr?
-      render json: {:message => "Response recorded" } and return
-    else
-      redirect_to course_path(@course) and return
+  
+    respond_to do |format|
```

Comment:
This block should probably only updating the page like this if the poll response successfully saves (and show an error message in the flash if not); the existing check, which should be here anyway, is on line 24 for reference